### DISCUSSION TITLE
Best parameters for low-energy electrons

### DISCUSSION THREAD

**Discussion by Michl5** [nrc-cnrc/EGSnrc#926] [Physics] [answered]
Hello, 
I am involved with depth dose distributions at low electron energy (20 to 100 keV). I am using DOSRZnrc with the Graphical User Interface (GUI), but I am not sure which settings, especially which transport parameters are the right ones. There are too many options and I don't want to try them all. Can anyone recommend me something? What are the important things to consider at such low energies?

Here are my choices so far:

```
Pair angular sampling - simple
Brems angular sampling - KM
Brems cross sections - NIST
Electron-step algorithm - PRESTA-II
Electron impact ionization - On
Boundary crossing algorithm - EXACT
Global ECUT - 0.512
Global PCUT - 0.001
Global SMAX - 1e10
Rayleigh scattering - On
Bound Compton - On
ECUT by regions - No
PCUT by regions - No
SMAX by Regions - No
PE angular sampling - On
Atomic relaxations - On
Photon cross section library - si
ESTEPE - 0.25
XImax - 0.5
Skin depth for BCA - 3
Spin effects - Yes
```

Thank you for all your recommendations.

**Comment by rtownson** [comment]
If you delete this section from the input file, it will use the defaults, which is usually recommended. ECUT and PCUT are the important ones to adjust for low energies, and you have them set properly at 1 keV. You have turned on some optional settings, which will have minor impact but are probably the right choice.

Generally I use the following, but in a depth dose I expect you will see no change:
```
Brems cross sections            = NRC
Photon cross sections           = xcom
```

**Comment by Michl5** [comment]
Thank you for quick answer.

**Comment by Michl5** [comment]
Looking to the .egslst file I noticed that the ECUT of 0.512 MeV set in the GUI was not used in the calculations. But the preset value of 0.521 MeV was used, obviously from the PEGS4 material data. Now finally I had understood why the material data are named 521icru, namely to indicate the energy cut. This may seem self-evident for many others, but I apologize for having to find my way into such contexts as a newcomer.  

Now I would like to ask and ask for help how I can realize the calculation with an ECUT of 0.512 MeV (1 keV kinetic)? Is there a data set for 512icru that I can download somewhere? Or is there another trick?

**Reply by rtownson** [reply]
The material data has the AE and AP fixed, but you should still be able to change the ECUT and PCUT. Here is an alternative, with all the same materials as 521icru, but where you can change AE and AP dynamically. Adjust the AP and UP as appropriate, and set global ECUT and PCUT in the transport parameters normally. Just add the following to the end of your input file, and then do not use "-p 521icru" when running the simulation.

```
:start media definition:
    ae = 0.512
    ue = 50.511
    ap = 0.001
    up = 50

material data file=/yourEGSpath/HEN_HOUSE/pegs4/data/material.dat

:stop media definition:
```

**Reply by YapingQi** [reply]
Hi RTownson, I came across the same question and added the media definition you suggested above, just replace the first line with ae = 0.01. But I got a segmentation fault.  Is the ae value too low?

 ****LOWEST ENERGY INPUT FOR DENSITY EFFECT IS 5.120E-01
                   WHICH IS HIGHER THAN THE VALUE OF AE= 1.000E-01 MEV
 ***IT HAS BEEN SET TO AE***

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.
Backtrace for this error:
#0  0x7f190433732a
#1  0x7f1904336503
#2  0x7f19039b3f0f
#3  0x5621f291b62f
#4  0x5621f291bab9
#5  0x5621f295fa81
#6  0x5621f2978bc6
#7  0x5621f299109f
#8  0x5621f28cd70e
#9  0x7f1903996c86
#10  0x5621f28cd739
#11  0xffffffffffffffff
Segmentation fault (core dumped)

**Reply by rtownson** [reply]
@YapingQi  The lowest valid AE is `ae = 0.512`, because it is total energy, not kinetic energy. For a `10 keV` threshold, use `ae = 0.521`

**Comment by Michl5** [comment, marked as answer]
Dear RTownson. Thank you for the tips. Since I am not so familiar with the programming structures and prefer to work with the graphical user interface, I have now found another way to set the Ecut to 0.512MeV. With the egs_gui.exe I could create a new PEGS4 material data collection by myself. Thereby I entered the values you suggested for ae, ue , ap and up. Then I could load this file (named as 512.icru.pegs4dat) by the GUI for RZ EGSnrc. This works very well now and I thank you for the help.

**Reply by rtownson** [reply]
Ok, I'm glad you got it working. Just keep in mind that ECUT is not set in the pegs4dat file, it is only set in your simulation parameters. So you can always change ECUT without changing the pegs4dat.
